### DISCUSSION TITLE
Compile errors for ZXing nuget object in a .net maui project

### DISCUSSION THREAD

**Discussion by wallym** [dotnet/maui#9178] [General]
I've got a project I started on a couple of weeks ago.  I've started noticing this weird behavior in VS.  I'm on 17.3 Preview 6 on windows11, but i've noticed this off and on with earlier previews as well.  I've added the ZXing scanner from Jon Dick.  I have the code below in my MauiProgram.cs file.  Vs is reporting the following errors below.  This had previously worked as recently as last weekend.  The error appears to be from the windows build.  I go into the project properties and I tell VS to not build for windows by unclicking the checkbox to build for windows, and VS immediately rechecks the checkbox.  I feel like I am stuck.  what is the correct process to not get this error besides unclick to build for windows?  If I have selected my attached android phone as the current debug target, why is it trying to build for windows?  What do I need to do on my end to not get this error and debug?  TIA

Severity	Code	Description	Project	File	Line	Suppression State
Error	CS0246	The type or namespace name 'ZXing' could not be found (are you missing a using directive or an assembly reference?)	ScanAndScoringMauiApp (net6.0-windows10.0.19041.0)	C:\SoftwareSource\ScanAndScoreClientMaui\ScanAndScoringMauiApp\MauiProgram.cs	1	Active

Severity	Code	Description	Project	File	Line	Suppression State
Error	CS1061	'MauiAppBuilder' does not contain a definition for 'UseBarcodeReader' and no accessible extension method 'UseBarcodeReader' accepting a first argument of type 'MauiAppBuilder' could be found (are you missing a using directive or an assembly reference?)	ScanAndScoringMauiApp (net6.0-windows10.0.19041.0)	C:\SoftwareSource\ScanAndScoreClientMaui\ScanAndScoringMauiApp\MauiProgram.cs	14	Active
 

[using ZXing.Net.Maui;

#if ANDROID
[assembly: Android.App.UsesPermission(Android.Manifest.Permission.Camera)]
#endif
namespace ScanAndScoringMauiApp;

public static class MauiProgram
{
	public static MauiApp CreateMauiApp()
	{
		var builder = MauiApp.CreateBuilder();
		builder
			.UseMauiApp<App>().UseBarcodeReader()
			.ConfigureFonts(fonts =>
			{
				fonts.AddFont("OpenSans-Regular.ttf", "OpenSansRegular");
				fonts.AddFont("OpenSans-Semibold.ttf", "OpenSansSemibold");
			});

		return builder.Build();
	}
}
](url)

**Comment by wallym** [comment]
It appears that this is a mystery.  I'm finding that closing VS and then reopening VS with the project in it seems to solve the problem.  I've run into similar problems where VS thinks there is no error but a compile shows an error which VS will then recognize and display in the error list.  Not an answer, just an observation.

**Comment by maxikraxi** [comment]
Hey, 
had the same Issue with Version "0.2.0 Preview 2", can you try to go back back to "0.1.0 Preview 5"?

**Reply by DavyDewit** [reply]
Thanks for this! i reinstalled 0.1.0 Preview 5 and this seemed to fix the issue!

**Comment by AndreasReitberger** [comment]
Have tried the latest version? Works fine on my end.
https://github.com/Redth/ZXing.Net.Maui

**Comment by SunilRana7121** [comment]
Hello Team,

I am facing below issue 
"CS1061	'MauiAppBuilder' does not contain a definition for 'UseBarcodeReader' and no accessible extension method 'UseBarcodeReader' accepting a first argument of type 'MauiAppBuilder' could be found (are you missing a using directive or an assembly reference?) "

And My Visual Studio configuration details :
Microsoft Visual Studio Professional 2022 (2) (64-bit) - Preview
Version 17.8.0 Preview 1.0

Please do the needful. Thank You in advance!!


Regards,
Sunil Rana
